sdv/metadata/single_table.py

```diff
@@ -242,6 +242,73 @@ def detect_from_csv(self, filepath, pandas_kwargs=None):
         data = pd.read_csv(filepath, **pandas_kwargs)
         self.detect_from_dataframe(data)
 
+    def set_primary_key(self, id):
+        """Set the metadata primary key.
+
+        Args:
+            id (str, tuple):
+                Name (or tuple of names) of the primary key column(s).
+        """
+        is_tuple_of_str = isinstance(id, tuple) and all(isinstance(i, str) for i in id)
+        if not isinstance(id, str) and not is_tuple_of_str:
+            raise ValueError('"primary_key" must be a string or tuple of strings.')
+
+        if self._metadata['primary_key'] is not None:
+            warnings.warn(
+                f"There is an existing primary key {self._metadata['primary_key']}."
+                ' This key will be removed.'
+            )
+
+        self._metadata['primary_key'] = id
+
+    def set_alternate_keys(self, ids):
+        """Set the metadata alternate keys.
+
+        Args:
+            ids (list[str], list[tuple]):
+                List of names (or tuple of names) of the alternate key columns.
+        """
+        is_list_of_tuple_str = all(isinstance(id, (str, tuple)) for id in ids)
+        is_list_of_tuple_of_str = \
+            all(all(isinstance(i, str) for i in id) for id in ids if isinstance(id, tuple))
+
+        if not isinstance(ids, list) or not is_list_of_tuple_str or not is_list_of_tuple_of_str:
+            raise ValueError(
+                '"alternate_keys" must be a list of strings or a list of tuples of strings.')
+
+        self._metadata['alternate_keys'] = ids
+
+    def set_sequence_key(self, id):
+        """Set the metadata sequence key.
+
+        Args:
+            id (str, tuple):
+                Name (or tuple of names) of the sequence key column(s).
+        """
+        is_tuple_of_str = isinstance(id, tuple) and all(isinstance(i, str) for i in id)
+        if not isinstance(id, str) and not is_tuple_of_str:
```

Comment:
we can probably pull this into a method that they can all use that checks if an id is a str or tuple of strings